tests/TestCase/Auth/BasicAuthenticateTest.php

```diff
@@ -214,12 +214,12 @@ public function testAuthenticateSuccess()
     /**
      * test scope failure.
      *
-     * @expectedException \Cake\Network\Exception\UnauthorizedException
-     * @expectedExceptionCode 401
      * @return void
      */
     public function testAuthenticateFailReChallenge()
     {
+        $this->expectException(\Cake\Network\Exception\UnauthorizedException::class);
+        $this->expectExceptionCode(401);
```

Comment:
Why is this notation an improvement? The annotation style is not deprecated and this type of change has the opportunity to create merge conflicts with the `3.next` branch.

Reply:
it is best practice and suggested way of expecting expectations since almost 2 years
(TLDR: code over annotation for better detection&refactoring, method call could be put directly before code that is expected to raise exception, not to whole codeblock, so better control of what is raising the exception)

Reply:
while merging into `3.next`, you could keep `3.next` version of code on each conflict and we could re-apply auto-changes in separated PR for `3.next`. that's easy ;)

Reply:
> method call could be put directly before code that is expected to raise exception

Thats when I would start putting it in code, and that needs to be done manually.

Reply:
that part of the benefit need to be achieved manually, yes.
having real class usage in code instead of text (annotation) is still benefit.

Reply:
Maybe this could be done in/for the 3.next branch instead.

Reply:
I sent this PR to default branch, if I shall change target branch, let me know ;)

Reply:
I will get this merged into master/3.next over the next few days. Any conflicts will be a one time cost.

Reply:
i'm not fully get it. it's already targeting master. shall I change PR target branch to sth ? (then I could solve the conflicts saving your time!)

Reply:
What you have now is fine. I can take care of any conflicts that come up when I merge master -> 3.next.